src/DynamoCore/Logging/Analytics.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using Dynamo.Models;
+
+namespace Dynamo.Logging
+{
+    /// <summary>
+    /// Utility class to support analytics tracking.
+    /// </summary>
+    public class Analytics
+    {
+        protected static IAnalyticsClient client = new DynamoAnalyticsClient();
+        protected static bool enabled = true;
+
+        /// <summary>
+        /// Starts the client when DynamoModel is created. This method initializes
+        /// the Analytics service and application life cycle start is tracked.
+        /// </summary>
+        /// <param name="model">DynamoModel</param>
+        /// <param name="enable">Whether to enable analytics and usage reporting.</param>
```

Comment:
It's just kind of a weird model. Calling `Start(false)`, is like saying, I want to create this object but I never want it to be used for anything. I realize that we're doing this to protect so that calls to the other analytics methods don't fail, but you could do the same by testing within those methods if `client==null`. Then you'd just never call `Start()` in test mode.